cmd/restic/cmd_forget.go

```diff
@@ -108,10 +108,16 @@ func runForget(opts ForgetOptions, gopts GlobalOptions, args []string) error {
 		return err
 	}
 
-	lock, err := lockRepoExclusive(gopts.ctx, repo)
-	defer unlockRepo(lock)
-	if err != nil {
-		return err
+	if gopts.NoLock && !opts.DryRun {
+		Warnf("Warning: --no-lock is ignored since it's only applicable in combination with --dry-run for forget command")
```

Comment:
I'd prefer to just error out, after all a plain `forget --no-lock` won't do what one expects.

Reply:
I've changed Warnf to Exitf